### DISCUSSION TITLE
Why does lima mount the host's home directory?

### DISCUSSION THREAD

**Discussion by marcoslar** [lima-vm/lima#627] [Q&A] [answered]
Hi all. So, perhaps it's a dumb question. Let me provide some context.

I have:

- defined `LIMA_HOME` as `/Users/myuser/opt/limahome` 
- provided a default config file in `$LIMA_HOME/_config/default.yaml` where I explicitly override the mount location:

```
mounts:
- location: "/Users/myuser/opt/limastuff/machine1"
```

But when executing `limactl start`, the logs print the following:

```
...
INFO[0157] [hostagent] The essential requirement 5 of 5 is satisfied
INFO[0157] [hostagent] Mounting "/Users/myuser/opt/limastuff/machine1"
INFO[0158] [hostagent] Mounting "/tmp/lima"
INFO[0158] [hostagent] Mounting "/Users/myuser"   <======================== Why?
INFO[0158] [hostagent] fuse: mountpoint is not empty
...
```

So, yes my custom mount location is being mounted but my host home directory is mounted as well. Why? 
I want to specify `writable: true` but only for my custom mount location, not for my host home directory.

**Comment by jandubois** [comment, marked as answer]
This is a limitation of the `default` and `override` mechanism: there is no way to remove an entry from a list; they just append, with the exception that `mounts` with the same `location` will update the other attributes instead of adding a duplicate entry.

You will have to edit the instance config itself if you want to remove the default mount points.

**Comment by marcoslar** [comment]
> This is a limitation of the default and override mechanism: there is no way to remove an entry from a list; they just append, with the exception that mounts with the same location will update the other attributes instead of adding a duplicate entry.

I understand. Is this by design or is it open to modification? 

> You will have to edit the instance config itself if you want to remove the default mount points.

I tried:

```
$ limactl stop default
$ limactl edit default
```

but the configuration file that is opened on my terminal only contains the mount location I originally defined (`/Users/myuser/opt/limastuff/machine1`) plus `/tmp/lima`. Not sure how I can remove the `~` mount location (I don't see it in the configuration of the instance). 

Thanks for replying that fast :) 

I copy here the mounts section of my configuration file:

```
...
# Expose host directories to the guest, the mount point might be accessible from all UIDs in the guest
# Default: null
mounts:
- location: "/Users/myuser/opt/limastuff/machine1"
  # CAUTION: `writable` SHOULD be false for the home directory.
  # Setting `writable` to true is possible, but untested and dangerous.
  # Default: false
  writable: null
  sshfs:
    # Enabling the SSHFS cache will increase performance of the mounted filesystem, at
    # the cost of potentially not reflecting changes made on the host in a timely manner.
    # Warning: It looks like PHP filesystem access does not work correctly when
    # the cache is disabled.
    # Default: true
    cache: null
    # SSHFS has an optional flag called 'follow_symlinks'. This allows mounts
    # to be properly resolved in the guest os and allow for access to the
    # contents of the symlink. As a result, symlinked files & folders on the Host
    # system will look and feel like regular files directories in the Guest OS.
    # Default: false
    followSymlinks: null
- location: "/tmp/lima"
  writable: true
  sshfs:
    followSymlinks: null
...
```

**Comment by marcoslar** [comment]
I tried again, and it worked 🤷 

One more question, is there any way to specify the directory in the guest on which the mount location will be mounted? So for instance,  I specify the mount location on my host to be `/Users/myuser/opt/limastuff/machine1` but on the guest the location could be `/machine1`.

**Reply by jandubois** [reply]
> is there any way to specify the directory in the guest

There is not, but it seems a reasonable request: https://github.com/lima-vm/lima/issues/130

**Comment by cpach** [comment]
@marcoslar @jandubois Hi folks! May I suggest that either one of you mark this question as answered? Thank you in advance!

**Comment by jandubois** [comment]
> is there any way to specify the directory in the guest on which the mount location will be mounted?

@marcoslar This is been implemented in lima 0.11.1 by #867
